### PR TITLE
Emmett bot

### PR DESCRIPTION
**Pull request by Emmett-Byrne**

### CODE REVIEW

**Comment by PrzemekTomczyk** [comment]
you're gonna have to wait until Cathal's screens are merged in

File: ARGO/include/AiSystem.h
Comment (CathalRedmond) [inline ARGO/include/AiSystem.h:40]: Is this a behaviour tree though?

File: ARGO/include/BehaviourTree.h
Comment (CathalRedmond) [inline ARGO/include/BehaviourTree.h]: 'DEFAUL' ?

Comment (CathalRedmond) [reply ARGO/include/BehaviourTree.h]: Also do you need to add bot to the start of all of these?

File: ARGO/include/BehaviourTree.h
Comment (CathalRedmond) [inline ARGO/include/BehaviourTree.h:59]: why not just have the 'child->getWeight()' here instead of 'weight'

File: ARGO/include/BehaviourTree.h
Comment (CathalRedmond) [inline ARGO/include/BehaviourTree.h:31]: These 3 structs share the same data they can probably be made into one struct

File: ARGO/include/BehaviourTree.h
Comment (CathalRedmond) [inline ARGO/include/BehaviourTree.h]: magic numbers 1 and 10

File: ARGO/include/BehaviourTree.h
Comment (CathalRedmond) [inline ARGO/include/BehaviourTree.h]: delete comment

Comment (CathalRedmond) [reply ARGO/include/BehaviourTree.h]: same on line 132

File: ARGO/include/BehaviourTree.h
Comment (CathalRedmond) [inline ARGO/include/BehaviourTree.h]: tertiary operator could be used here

File: ARGO/src/AiSystem.cpp
Comment (CathalRedmond) [inline ARGO/src/AiSystem.cpp]: is commented needed

File: ARGO/src/Game.cpp
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: Too many magic numbers

File: ARGO/src/Game.cpp
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: remove this comment

File: ARGO/src/Game.cpp
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: imp?

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/include/BehaviourTree.h
Comment (PrzemekTomczyk) [reply ARGO/include/BehaviourTree.h]: At least have a comment explaining wtf are you doing here @NerakaiStudent 👍

**Review by PrzemekTomczyk — CHANGES_REQUESTED** [review]

File: ARGO/src/GameScreen.cpp
Comment (CathalRedmond) [inline ARGO/src/GameScreen.cpp]: is commented code needed?

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

**Pull request merged**
